diff --git a/node_modules/@chainsafe/libp2p-yamux/dist/src/decode.js b/node_modules/@chainsafe/libp2p-yamux/dist/src/decode.js
index 58728fddc7355906bcb8d8273d41e73ecbbc9d67..7aa39de8fc9268d632be340d10f87a2522f18c45 100644
--- a/node_modules/@chainsafe/libp2p-yamux/dist/src/decode.js
+++ b/node_modules/@chainsafe/libp2p-yamux/dist/src/decode.js
@@ -111,14 +111,14 @@ export class Decoder {
 export function returnlessSource(source) {
     if (source[Symbol.iterator] !== undefined) {
         const iterator = source[Symbol.iterator]();
-        iterator.return = undefined;
+        Object.defineProperty(iterator, 'return', {});
         return {
             [Symbol.iterator]() { return iterator; }
         };
     }
     else if (source[Symbol.asyncIterator] !== undefined) {
         const iterator = source[Symbol.asyncIterator]();
-        iterator.return = undefined;
+        Object.defineProperty(iterator, 'return', {});
         return {
             [Symbol.asyncIterator]() { return iterator; }
         };
